### PR TITLE
feat: add prometheus support for reporting ui

### PR DESCRIPTION
**Pull request by rdmulford**

### CODE REVIEW

File: packages/client/src/metricSources/Prometheus/PrometheusMetricModal.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            this.touch('customInlineTemplate');
+          }}
+          touched={this.state.touched.customInlineTemplate}
+          error={this.state.errors['query.customInlineTemplate']}
+          id="customInlineTemplate"
+          label="Custom Inline Template"
+          value={this.state.metric.query.customInlineTemplate}
+          onChange={e => this.updateQueryObject('customInlineTemplate', e.target.value)}
+          placeHolderText="IDK WHAT THIS IS LMAO"
+          subText="boy i sure hope this works"
```
Comment (fieldju) [inline packages/client/src/metricSources/Prometheus/PrometheusMetricModal.tsx]: This should be help text that shows up under the text box.

I see though that you are implementing this slightly different than we did with SignalFx so probably need to fire this up locally to sanity check.

https://github.com/Nike-Inc/referee/blob/master/packages/client/src/metricSources/SignalFx/SignalFxMetricModal.tsx#L173-L205

**Comment by melanahammel** [comment]
Hi @rdmulford, I just ran your PR locally to test it. It does not look like validation is wired up.
For SignalFx, if users do not provide a custom inline template, we show this error:
image

The existing validation is in [this file](https://github.com/Nike-Inc/referee/blob/feat/enable_in_progress_reports/packages/client/src/validation/configValidators.ts#L124:L139). There is some level of complexity because we decide whether or not to run that validation based on the boolean `isQueryTypeSimple` because there are two different ways to provide queries in SignalFx, simple and a custom inline template.

Currently, you are sending `isQueryTypeSimple` as `true` [here](https://github.com/rdmulford/referee/blob/prometheus/packages/client/src/metricSources/Prometheus/PrometheusMetricModal.tsx#L12). If you switch that to `false` (because you are using a custom inline template and not a simple query), then the validation should work.

Additionally, the validation is SignalFx specific as you can see [here](https://github.com/Nike-Inc/referee/blob/feat/enable_in_progress_reports/packages/client/src/validation/configValidators.ts#L89:L101). Because we bring in `type`, you could add Prometheus specific validation. 

What is the current behavior when you send through an empty custom inline template for a metric?
Let me know if you have any more questions about the validation.

File: packages/client/src/stores/ConfigEditorStore.ts [hunk 2]
```diff
@@ -96,8 +96,12 @@ export default class ConfigEditorStore {
 
   @action.bound
   setCanaryConfigObject(canaryConfigObject: CanaryConfig): void {
-    this.canaryConfigObject = canaryConfigObject;
-    canaryConfigObject.metrics.forEach(metric => metric.groups.forEach(g => this.createNewGroup(g, false)));
+    if (canaryConfigObject.metrics == null) {
+      this.canaryConfigObject = canaryConfigObject.canaryConfig
```
Comment (melanahammel) [inline packages/client/src/stores/ConfigEditorStore.ts]: `canaryConfig` does not exist on `canaryConfigObject` (see [here](https://github.com/Nike-Inc/referee/blob/master/packages/client/src/domain/Kayenta.ts#L118:L132)).

What is the purpose of this conditional statement?

**Comment by rdmulford** [comment]
> Hi @rdmulford, I just ran your PR locally to test it ...

Hey @melanahammel thanks for taking a look at this. To give you some more context on our use-case, currently we have scripts in place to output a link to the Kayenta result json, but want to use referee to display that output nicely instead of having to look through the raw json blob. For this we just need prometheus to be a valid metric source type so that this line does not throw an error: https://github.com/Nike-Inc/referee/blob/850924e66c775a0555560803ca94ba54ea9a1ca2/packages/client/src/components/shared/IndividualMetricView.tsx#L170

So this PR just addresses adding prometheus as a metric source type and ensuring that nothing crashes for the prometheus source type. We weren't really looking to add full prometheus configuration support, rather just support for metrics using prometheus to be viewable by the reporting ui without crashing.

Please let me know if you have any concerns, thanks.

**Comment by melanahammel** [comment]
Hi @rdmulford, thanks for the context on what you are trying to accomplish.

Based on the pattern established by the other two metric sources we have integrated into Referee, Prometheus is very close to full configuration support. In order to get to full configuration support, I see two additional changes:

- [ ]  adding validation to `customInlineTemplate`
- [ ]  fixing [this line](https://github.com/Nike-Inc/referee/pull/61/files/ecac8f3dc5fe8fa432ac9cefc271d82d62b94133#diff-3ef60c2e6fa69a555a9d8a7ebc424f93) that is trying to access an object that does not exist

I understand the scope of your use case is slightly narrower, but for those who will want to use Prometheus in the future, it's better for the community if it is fully set up and supported from the beginning when we add the metrics source. We are more than willing to work with you to add the full support if you have any questions.

**Comment by rdmulford** [comment]
> Hi @rdmulford, thanks for the context ...

I understand and this seems reasonable, I will take a stab at completing the two remaining steps in order to achieve full prometheus support. Thanks! 😄

File: packages/client/src/stores/ConfigEditorStore.ts (hunk 2 above)
Comment (rdmulford) [reply packages/client/src/stores/ConfigEditorStore.ts]: So we post to the `/standalone_canary_analysis` endpoint of our standalone Kayenta API to have it start monitoring and for that our json needed to be structured with our configuration in an object called `canaryConfig`, but it seems like referee expects the config to be flat, so this was sort of a quick fix to make that work. What would be the best way to make the configuration work for both?

**Pull request merged**